### ISSUE TITLE
Comma rule doesn't trigger in if or guard statements

### ISSUE THREAD

**Issue by johnnygao** [realm/SwiftLint#2825]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

It seems that the Comma rule doesn't trigger in if or guard statements where the conditions are separated with a comma.

### Environment

* SwiftLint 0.34.0
* Installation method used: Homebrew
* No config file
* Xcode version 10.3 (10G8)
* Sample:
```swift
class EditAddressViewController: UITableViewController {
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        guard let cell = tableView.dequeueReusableCell(withIdentifier: "fieldInputCell") as? FieldInputCell ,
            mode = viewModel?.formConfig[indexPath.row] else { return UITableViewCell() }
        cell.configure(mode: mode)
        return cell
    }
}
```

In this sample, I'd expect the rule to trigger on line 3, which it currently doesn't. It does if there is another space after the comma, but not if the comma is followed by a newline.
